onboard/nav/stateMachine.cpp

```diff
@@ -341,18 +387,47 @@ NavState StateMachine::executeDrive()
         {
             return NavState::SearchSpin;
         }
-        mPhoebe->roverStatus().path().pop();
+        mPhoebe->roverStatus().path().pop_front();
+        if (mPhoebe->roverStatus().currentState() == NavState::RadioRepeaterDrive)
+        {
+            return NavState::RepeaterDropWait;
+        }
         ++mCompletedWaypoints;
         return NavState::Turn;
     }
     if( driveStatus == DriveStatus::OnCourse )
     {
+        if (mPhoebe->roverStatus().currentState() == NavState::RadioRepeaterDrive)
+        {
+            return NavState::RadioRepeaterDrive;
+        }
         return NavState::Drive;
     }
     // if driveStatus == DriveStatus::OffCourse
```

Comment:
NIT: This comment is now confusing with the new logic below, could we make the comment say else so it's less misleading. Or remove it altogether. As is, it looks like it goes with the if statement and that's not true.